de.metas.acct.base/src/main/sql/postgresql/ddl/functions/AccountSheetReport.sql

```diff
@@ -0,0 +1,238 @@
+DROP FUNCTION IF EXISTS AccountSheetReport(p_dateFrom date, p_dateTo date, p_c_acctschema_id NUMERIC, p_ad_org_id numeric, p_account_id NUMERIC, p_c_activity_id numeric, p_c_project_id numeric);
+
+/*
+- DateFrom/To - mandatory
+- C_AcctSchema_ID - mandatory
+- AD_Org_ID - mandatory
+- Account_ID - optional
+- C_Activity_ID - optional
+- C_Project_ID - optional
+*/
+CREATE OR REPLACE FUNCTION AccountSheetReport(p_dateFrom date,
+                                              p_dateTo date,
+                                              p_c_acctschema_id NUMERIC,
+                                              p_ad_org_id numeric,
+                                              p_account_id NUMERIC=NULL,
+                                              p_c_activity_id numeric=NULL,
+                                              p_c_project_id numeric=NULL)
+    RETURNS table
+            (
+                dateacct         timestamp WITHOUT TIME ZONE,
+                documentno       text,
+                description      text,
+                amtacctdr        numeric,
+                amtacctcr        numeric,
+                account_id       numeric,
+                fact_acct_id     numeric,
+                c_doctype_id     numeric,
+                c_tax_id         numeric,
+                c_taxcategory_id numeric,
+                beginningBalance numeric,
+                endingBalance    numeric
+            )
+AS
+$BODY$
+DECLARE
+    v_time                              timestamp;
+    LINE_TYPE_BEGINNINGBALANCE CONSTANT text = 'B';
+    LINE_TYPE_TRANSACTION      CONSTANT text = 'T';
+    v_temp                              numeric;
+BEGIN
+    v_time := logDebug('start');
+
+    --
+    -- create temporary table for everything we're working on; it has no rows, only the needed columns
+    DROP TABLE IF EXISTS TMP_AccountSheetReport;
+    CREATE TEMPORARY TABLE TMP_AccountSheetReport AS
+        (
+            SELECT fa.*,
+                   tc.c_taxcategory_id,
+                   0::numeric beginningBalance,
+                   0::numeric endingBalance,
+                   NULL::text lineType
+            FROM c_elementvalue ev
+                     INNER JOIN fact_acct fa ON ev.c_elementvalue_id = fa.account_id -- only search for accounts with at least 1 transaction
+                     LEFT JOIN c_tax t ON fa.c_tax_id = t.c_tax_id
+                     LEFT JOIN c_taxcategory tc ON t.c_taxcategory_id = tc.c_taxcategory_id
+            WHERE FALSE
+        );
```

Comment:
just create the table with precisely the columns u need; avoid using this "tricky" SELECT

Reply:
done;